jasypt/src/main/java/org/jasypt/util/YamlEventHandlers/EventEncryptor.java

```diff
@@ -0,0 +1,20 @@
+package org.jasypt.util.YamlEventHandlers;
+
+import java.util.Properties;
+
+import org.jasypt.intf.cli.JasyptEncryptorUtil;
+import org.yaml.snakeyaml.events.Event;
+import org.yaml.snakeyaml.events.ScalarEvent;
+
+public class EventEncryptor {
+	
+	public Event encryptValueInScalarEvent(Event event, Properties argumentValues, JasyptEncryptorUtil encryptor) {
+		String inputValue = ((ScalarEvent) event).getValue();
+		if (inputValue.length() == 0) return event;
+		String encryptedValue = encryptor.encrypt(inputValue);
+		String encryptedValueWrapped = "ENC(" + encryptedValue + ")";
+		Event newEvent = YamlUtil.getScalarEventFromString(encryptedValueWrapped);
+//		((ScalarEvent) event).setValue(encryptedValueWrapped);   // TODO This would've been ideal. Try to improve above line to achieve this.
```

Comment:
@prakashiit Is this TODO still relevant?

Reply:
@SriMouli Yes, if someone would be interested in improving this code. The current code isn't perfect but it would suffice for our maximum use cases. The commented out code would've been the perfect solution but it isn't allowed, my code is just a workaround.